lib/stale.js

```diff
@@ -80,7 +80,68 @@ module.exports = class Stale {
 
     const query = queryParts.join(' ')
     const days = this.getConfigValue(type, 'days') || this.getConfigValue(type, 'daysUntilStale')
-    return this.search(type, days, query)
+    const results = await this.search(type, days, query)
+
+    const pinned = await this.getPinnedNumbers(type)
+    const nonPinned = results.data.filter(issue => pinned.includes(issue.number))
+    return {
+      ...results,
+      data: {
+        ...results.data,
+        total_count: nonPinned.length,
+        items: nonPinned,
+      }
+    }
+  }
+
+  async getPinnedNumbers(type) {
+    if (type === 'pulls') {
+      // Pull Requests cannot be pinned
+      return []
+    } else if (type !== 'issues') {
+      throw new Error(`Unknown type: ${type}. Valid types are 'pulls' and 'issues'`)
+    }
+
+    const { owner, repo } = this.config
+
+    try {
+      // GitHub's v3 REST API doesn't support Pinned Issues; v4 GraphQL API does
+      const {data, errors} = await this.github.graphql(
+        `
+          query Issues($owner: String!, $repo: String!) {
+            repository(owner: $owner, name: $repo) {
+              pinnedIssues(first: 100) {
+                nodes {
+                  number
+                }
+                pageInfo {
+                  # This should always be false, as only 3 issues can be pinned at the same time
+                  hasNextPage
+                }
+              }
+            }
+          }
+        `,
+        {
+          owner,
+          repo,
+          headers: {
+            // Opt-in to Pinned Issues API preview
+            accept: 'application/vnd.github.elektra-preview+json'
+          },
+        },
+      )
+
+      if (errors && errors.length) {
+        throw new Error(errors[0].message)
+      }
+
+      return data.repository.pinnedIssues.nodes.map(issue => issue.number)
+    } catch (error) {
+      this.logger.error(`Encountered an error while excluding pinned items for ${owner}/${repo}: ${error}`)
```

Comment:
I wasn't sure if this was the best way to log errors, but I think it's important that the bot not break if the _Pinned Issues Preview API` changes.